src/valkey/js_valkey.zig

```diff
@@ -148,6 +339,101 @@ pub const JSValkeyClient = struct {
         });
     }
 
+    /// Clone this client while remaining in the initial disconnected state. This does not preserve
+    /// the
+    pub fn cloneWithoutConnecting(this: *const JSValkeyClient) *JSValkeyClient {
+        const vm = this.globalObject.bunVM();
+
+        // Make a copy of connection_strings to avoid double-free
+        const connection_strings_copy: []u8 = if (this.client.connection_strings.len > 0) blk: {
+            const duped = bun.default_allocator.dupe(u8, this.client.connection_strings) catch break :blk &[0]u8{};
+            break :blk @constCast(duped);
+        } else &[0]u8{};
```

Comment:
```suggestion
        // Make a copy of connection_strings to avoid double-free
        const connection_strings_copy = try bun.default_allocator.dupe(u8, this.client.connection_strings);
```
and also change return type to `bun.OOM!*JSValkeyClient`. the zig allocator interface checks for allocating 0 bytes